doc/memos/rdm0.md

```diff
@@ -0,0 +1,126 @@
+
+In order to facilitate RIOT maintenance in the long term and at large scale, 
+additional documentation complementing code and usual code documentation is 
+needed. For instance, such memos are expected to describe and give explanations 
+about architectural design decisions, code structure etc. Other memos might also
+describe other aspects of RIOT activity, including, but not limited to, RIOT 
+community processes, position with respect to some related external technical 
+debates etc.
+
+
+This memo specifies a format for such memos and the process of discussing their 
+content before publication. Format and processes are somewhat inspired by 
+[IETF](http://ietf.org/) processes and documents such as Request For Comments 
+(RFC).
+
+# 2. RIOT Developers Memo Format
+
+RIOT Developer Memos (RDM) are written in English, using the Markdown (.md) 
+format. Each memo has a unique identification number indicated in its header, 
+e.g. this memo is RDM0.
+
+The structure of a memo should contain:
+
+1. a preamble as defined below,
+2. the content of the memo structured as needed,
+3. a postamble as defined below.
+
+The preamble of the memo should be structured as such:
+
+1. a header formated as the header of this memo,
+2. a short abstract,
+3. a status for the memo stating its license and the category of the memo (e.g. 
+that it is a consensus document),
+4. optionally, a terminology section which recalls/disambiguates definition of 
+terms that are key to the memo.
+
+The core content of the memo is structured as deemed necessary. However, during 
+the review process, care must be taken to follow the below principles as much as
+possible:
+
+1. one memo = one problem/aspect,
+2. start by crisply explaining and motivating the one problem/aspect the memo 
+targets,
+3. memos are expected to be of normative and durable nature, hence: keep a memo 
+as concise as possible without impairing clarity, and factor out content that 
+would be likely to be updated too often.
+
+The postamble of the memo should be structured as such:
+
+1. (optional) acknowledgements section thanking contributors other than the 
+authors,
+2. a section with the main references that relate to the content, with a strong
+preference for permanent, self-contained references.
+
+
+# 3. Memo Publishing and Maintenance Process
+
+Each memo is proposed as a pull-request (PR) and discussed the same way RIOT
+code PRs are processed on GitHub.
+
+
+Once proposed, a memo must have received an ACK by at least 3 maintainers other 
+than the author(s) of the memo, before its publication. 
+
+
+Once merged into the master branch of RIOT, a memo is considered published.
+
+Once merged, small changes/clarifications may be subsequently PRed to improve a
+memo. Bigger changes should result in a new memo deprecating the old memo.
+
+A dedicated directory in RIOT codebase (RIOT/doc/memos) contains all the memos.
+It is acceptable to add graphical content (in RIOT/doc/memos/graphics/) to
```

Comment:
I think it would be worth considering either including the memo number in the name of any graphical media under RIOT/doc/memos/graphics/ or include sub directories. The motivation is that if pictures are reused but get changed at some point it can be checked they still are correct in their context. (Although we want to keep use to a minimum.)

Reply:
Can I propose a separate repo for the RDMs? This would give them a more obvious presence for new contributors. If they're in a sub-folder of the main repo, _we_ may know they are there, but less people will spot them if they don't know about them. 
Also, the git histories will be separate, which makes it easier to get an overview of the RDM history, and the PRs and issues will be separated out, etc. In general, it would be a clearer, neater way of organization and easier to maintain.
Re. Doxygen, we could generate a website from this separate repo upon pull to master and deploy it at doc.riot-os.org/rdm, with a link on the main page or the sidebar. I.e., it doesn't seem to have too much overhead when considering Doxygen.

Reply:
could also use Travis + GitHub pages to automatically generate a nice HTML version - I guess?

Reply:
regarding directory structure, why not have a more generic `files` subfolder with subfolders per RDM (folder name = rdmXXXX)  and common folders for shared files. I think there might be other things than graphics (e.g. code snippets, style sheets for rendering) only that can be stored there.